src/helpers.cpp

```diff
@@ -111,6 +111,28 @@ std::vector<JointParameter> loadJointsFromParameter(
   return output;
 }
 
+std::unordered_map<std::string, double> loadGoalConstraintsFromParameter(
+      ros::NodeHandle& nodeHandle,
```

Comment:
This should be `const ros::NodeHandle&`

Reply:
In `ada_demos`, you and @jslee02 told me that NodeHandles, shouldn't be references typically, so I corrected it in the new commit.

Reply:
This case is different from the `ada_demos` case; The classes in `ada_demo` take the ownership but not this function. In this case, we should prefer passing by const reference to avoid unnecessary reference counting.